### ISSUE TITLE
new-parens: add ability to allow chained prop/method access

### ISSUE THREAD

**Issue by ZenVoich** [eslint/eslint#12590]
**What rule do you want to change?**
new-parens

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

```js
let year = new Date().getFullYear();
```

**What does the rule currently do for this code?**
Displays an error

**What will the rule do after it's changed?**
Allow this code `new Date().getFullYear()` and disallow this code `new Date()`

**Are you willing to submit a pull request to implement this change?**
No

**Comment by mdjermanovic** [comment]
Hi @ZenVoich, thanks for the proposal!

I guess you would like an exception to the `"never"` option? And that's because removing the parens in this case requires adding necessary outer parens, and `(new Date).getFullYear()` doesn't look any better than the original `new Date().getFullYear()` ?

**Comment by ZenVoich** [comment]
> I guess you would like an exception to the "never" option?

Yes

> And that's because removing the parens in this case requires adding necessary outer parens, and (new Date).getFullYear() doesn't look any better than the original new Date().getFullYear() ?

I didn't think about the first option. I think this can be used as a workaround at the moment. But I'd prefer the second)

**Comment by mdjermanovic** [comment]
This makes sense to me. An option to allow only necessary parens (those that wouldn't require adding outer parens instead), either as an additional modifier for the `"never"` option or maybe as a new value for the string option (something like `"as-needed"`).

I left :+1: for the proposal. A rule enhancement needs 3 votes from the team and a champion to be accepted.

**Comment by ZenVoich** [comment]
@mdjermanovic Thanks!

**Comment by mdjermanovic** [comment]
I'll champion and I could work on this. Removed my vote to avoid confusion, so this now needs two more :+1: from the team to be accepted.

In favor of the proposal, the same code which would be added for this enhancement can be also used to improve the fixer for the existing behavior. In particular, to avoid unnecessary parens in the autofix when the option is `"never"`. In the actual version, the fixer always wraps the `new` expression in parens. 

For example, `const foo = new Bar();` becomes `const foo = (new Bar);` like in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5ldy1wYXJlbnM6W1wiZXJyb3JcIiwgXCJuZXZlclwiXSAgKi9cblxuY29uc3QgZm9vID0gbmV3IEJhcigpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OCwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Comment by aladdin-add** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
